ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/FutureAssertions.scala

```diff
@@ -34,13 +40,126 @@ final class FutureAssertions[T](future: Future[T]) {
       case Success(value) => Failure(new ExpectedFailureException(context, value))
       case Failure(other) => Failure(other)
     }
+
 }
 
 object FutureAssertions {
 
+  def assertAfter[V](
+      delay: FiniteDuration
+  )(test: => Future[V]): Future[V] = {
+    Delayed.Future.by(delay)(test)
+  }
+
+  def succeedsEventually[V](
+      delay: FiniteDuration = 100.millis,
+      maxInterval: FiniteDuration,
+      description: String,
+  )(
+      test: => Future[V]
```

Comment:
Is this actually called twice or is it cached after the first call?

I did a bit of local testing and it seems fine, but I just want to make sure.

Reply:
It will be called up to `maxInterval / delay`, as we need to retry until it either succeeds or the interval expires